### DISCUSSION TITLE
merge into master branch more often

### DISCUSSION THREAD

**Discussion by emiliom** [OSOceanAcoustics/echopype#227] [Ideas]
I spent a good chunk of time with Don's echopype PR, though in reality the bulk of my time was spent wrapping my head around class redesign changes. I noticed that it looks like you're not using the master branch for current stable-ish code. Instead, the `class-redesign` branch is accumulating tons of new, large changes. 

For transparency and convenience, I would encourage you (@leewujung and @ngkavin ) to move to the model of pushing to master often, basically all changes that are reasonably stable. In this model (I'm lazy and won't include a link that describes the git/github models), development branches never get too far ahead of master, and master always represents the main current development version that generally works. I'm happy to discuss this.

**Comment by leewujung** [comment]
Thanks. I actually do know the best practices recommendations, but it is not easy to implement when there are lots of other competing (funded) priorities. ;)   I am particularly hesitant about merging class-redesign to master before I have a chance to go through everything myself -- I don't think the code are reasonably stable -- which will happen during the upcoming break.  

I can see doing this once the class-redesign is merged though, since my motivation for the overhaul was exactly to reduce the time to maintain the code.
